src/Forms/TreeDropdownField.php

```diff
@@ -519,13 +520,20 @@ public function tree(HTTPRequest $request)
 
         // Allow to pass values to be selected within the ajax request
         $value = $request->requestVar('forceValue') ?: $this->value;
-        if ($value && ($values = preg_split('/,\s*/', $value ?? ''))) {
+        if ($value instanceof SS_List) {
+            $values = $value;
+        } elseif ($value) {
+            $values = preg_split('/,\s*/', $value ?? '');
+        } else {
+            $values = [];
+        }
+        if (!empty($values)) {
             foreach ($values as $value) {
                 if (!$value || $value == 'unchanged') {
                     continue;
                 }
 
-                $object = $this->objectForKey($value);
+                $object = is_object($value) ? $value : $this->objectForKey($value);
```

Comment:
This happened when the `SS_List` was being iterated over. I think it was a `DataList` specifically in the scenario I had.
No need to fetch the object if we already have it.

Reply:
I've lost context over this change but given I'm talking about `SS_List` above it's probably some subclass of `ModelData`.